lib/node-gyp.js

```diff
@@ -166,10 +167,18 @@ proto.spawn = function spawn (command, args, opts) {
   if (!opts) {
     opts = {}
   }
+
+  var cp = childProcess.spawn(command, args, opts)
+
   if (!opts.silent && !opts.stdio) {
-    opts.stdio = [0, 1, 2]
+    cp.stdout.pipe(split()).on('data', function (line) {
```

Comment:
This is probably OK but it's nice to make a habit of avoiding using `'data'` events so you can set up proper backpressure (not really a requirement here because npmlog doesn't give us a mechanism for it), usually done by implementing a custom stream.

maybe something like this

```js
const { Transform } = require('stream')

...

function outLogger (prefix, log) {
  return new Transform({
    transform: (chunk, enc, callback) {
      log(prefix, chunk.toString())
      callback()
    }
  })
}

cp.stdout.pipe(split()).pipe(outLogger('spawn stdout', log.notice)) // does it need to be `log.notice.bind(log)`?
cp.stderr.pipe(split()).pipe(outLogger('spawn stderr', log.error))
```

Reply:
Hmmm. I was wondering why you were suggesting this route in the original issue. I hadn't heard of a reason to not just use the `data` event. I didn't see an advantage to using a Transform stream but makes sense.

To be clear, this is the `data` event from `split()` and not from `stdout`/`stderr`.

I wonder if there are any possible improvements from using a `Writable` stream instead of a `Transform` since there is no data coming out of the logger (as far as the stream api is concerned).

Reply:
One problem that I see with this is that the subprocess loses the "TTY-ness" which many programs use to determine whether or not to out put output colors to the terminal.

Reply:
@TooTallNate This seems like a reason to add an "opt-in" option.